ams/ResetCommand.cs

```diff
@@ -32,67 +32,87 @@ internal class ResetCommand : BaseMigrator
+                var containers = storageClient.GetBlobContainersAsync(
+                               cancellationToken: cancellationToken);
+                _logger.LogInformation("The total containers count of the storage account is {count}.", totalItems);
+                List<BlobContainerItem>? filteredList = await containers.ToListAsync();
+                int resetAssetCount = 0;
+                foreach (var container in filteredList)
+                {
+                    BlobContainerClient containerClient = storageClient.GetBlobContainerClient(container.Name);
+                    await MigrateTaskAsync(containerClient, resetAssetCount, cancellationToken);
+                }
             }
-            catch (Exception)
+            else if (account != null)
+            {
+                _logger.LogInformation("Begin reset assets on account: {name}", account.Data.Name);
+                await _resourceProvider.SetStorageResourceGroupsAsync(account, cancellationToken);
+                AsyncPageable<MediaAssetResource> assets = account.GetMediaAssets()
+                    .GetAllAsync(cancellationToken: cancellationToken);
+                List<MediaAssetResource>? assetList = await assets.ToListAsync(cancellationToken);
+                int resetAssetCount = 0;
+                foreach (var asset in assetList)
+                {
+                    var (storage, _) = await _resourceProvider.GetStorageAccount(asset.Data.StorageAccountName, cancellationToken);
+                    var container = storage.GetContainer(asset);
+                    if (!await container.ExistsAsync(cancellationToken))
+                    {
+                        _logger.LogWarning("Container {name} missing for asset {asset}", container.Name, asset.Data.Name);
+                        return;
+                    }
+
+                    await MigrateTaskAsync(container, resetAssetCount, cancellationToken);
+                }
+                _logger.LogDebug("{resetAssetCount} out of {totalAssetCount} assets has been reset.", resetAssetCount, assetList.Count);
+            }
+            else
             {
                 _logger.LogError("No valid media account was found.");
                 throw new Exception("No valid media account was found.");
             }
-            _logger.LogInformation("Begin reset assets on account: {name}", account.Data.Name);
-            await _resourceProvider.SetStorageResourceGroupsAsync(account, cancellationToken);
-            AsyncPageable<MediaAssetResource> assets = account.GetMediaAssets()
-                .GetAllAsync(cancellationToken: cancellationToken);
-            List<MediaAssetResource>? assetList = await assets.ToListAsync(cancellationToken);
-            int resetAssetCount = 0;
-            foreach (var asset in assetList)
+        }
+        protected async Task MigrateTaskAsync(BlobContainerClient container, int resetAssetCount, CancellationToken cancellationToken)
```

Comment:
Two issues on this function:

(1)  param  int resetAssetCount should be changed to "ref int resetAssetCount", because this value is updated inside the function,  and it should be reported back to the caller.

(2) protected  can be changed to private,  there is no derived class from ResetCommand.